src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/applyforcosts/ApplyForCostsHandler.java

```diff
@@ -0,0 +1,106 @@
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackStage;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.Document;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.IdValue;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.YesOrNo;
+import uk.gov.hmcts.reform.iacaseapi.domain.handlers.PreSubmitCallbackHandler;
+import uk.gov.hmcts.reform.iacaseapi.domain.service.ApplyForCostsAppender;
+
+@Component
+public class ApplyForCostsHandler implements PreSubmitCallbackHandler<AsylumCase> {
+    private ApplyForCostsAppender applyForCostsAppender;
+
+    public ApplyForCostsHandler(ApplyForCostsAppender applyForCostsAppender) {
+        this.applyForCostsAppender = applyForCostsAppender;
+    }
+
+    @Override
+    public boolean canHandle(PreSubmitCallbackStage callbackStage, Callback<AsylumCase> callback) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        return callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
+                && callback.getEvent() == Event.APPLY_FOR_COSTS;
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<AsylumCase> handle(PreSubmitCallbackStage callbackStage, Callback<AsylumCase> callback) {
+        if (!canHandle(callbackStage, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        final AsylumCase asylumCase = callback.getCaseDetails().getCaseData();
+
+        String typeOfAppliedCosts = asylumCase.read(APPLIED_COSTS_TYPES, DynamicList.class)
+                .orElseThrow(() -> new IllegalStateException("typesOfAppliedCosts is not present"))
+                .getValue()
+                .getLabel();
+
+        Optional<String> argumentsAndEvidenceDetails = asylumCase.read(ARGUMENTS_AND_EVIDENCE_DETAILS, String.class);
+
+        Optional<List<IdValue<Document>>> argumentsAndEvidenceDocuments = asylumCase.read(ARGUMENTS_AND_EVIDENCE_DOCUMENTS);
+
+        Optional<List<IdValue<Document>>> scheduleOfCostsDocuments = asylumCase.read(SCHEDULE_OF_COSTS_DOCUMENTS);
+
+        YesOrNo applyForCostsHearingType = asylumCase.read(APPLY_FOR_COSTS_HEARING_TYPE, YesOrNo.class)
+                .orElseThrow(() -> new IllegalStateException("applyForCostsHearingType is not present"));
+
+        String applyForCostsHearingTypeExplanation = "";
+
+        if (applyForCostsHearingType.equals(YesOrNo.YES)) {
+            applyForCostsHearingTypeExplanation = asylumCase.read(APPLY_FOR_COSTS_HEARING_TYPE_EXPLANATION, String.class)
+                    .orElseThrow(() -> new IllegalStateException("applyForCostsHearingTypeExplanation is not present"));
+        }
+
+        String legalRepName = asylumCase.read(LEGAL_REP_NAME, String.class)
+                .orElseThrow(() -> new IllegalStateException("legalRepName is not present"));
```

Comment:
Either this logic or this handler should be protected so that Internal journey doesnt access this.

Reply:
Added check for LegalRep journey to the condition